src/main/java/hudson/plugins/gradle/injection/GradleEnterpriseExceptionTaskListenerDecoratorFactory.java

```diff
@@ -51,7 +51,7 @@ public GradleEnterpriseExceptionTaskListenerDecorator(Run run) {
         @Nonnull
         @Override
         public OutputStream decorate(@Nonnull OutputStream logger) {
-            if (isBuildAgentErrorsEnabled()) {
+            if (run != null) {
```

Comment:
The fix. We cannot call `isBuildAgentErrorsEnabled` from the agent side, and even if this value were computed on the controller side and serialized, it would not help because `run` would still have been left null by deserialization and could not be used here. So simply disable the decorator when remoted.

Reply:
Could we use a more expressive condition to "know" we're on the controller ? Maybe `jenkins.util.JenkinsJVM#isJenkinsJVM` ? 
As an additional safety net, I think we also need to combine the condition with `isBuildAgentErrorsEnabled()` (so only executed on the controller) even if it's already called upfront in `hudson.plugins.gradle.injection.GradleEnterpriseExceptionTaskListenerDecoratorFactory#of`. 
Btw, `org.jenkinsci.plugins.workflow.log.TaskListenerDecorator.Factory#of` implementations are executed only on the controller, right?

Reply:
> Could we use a more expressive condition to "know" we're on the controller ? Maybe `jenkins.util.JenkinsJVM#isJenkinsJVM` ?

We could, though it should not matter—if running on an agent, certainly `run == null`. A `checkJenkinsJVM` call could be added (inside the `run != null` block) as a matter of documentation, though it would perhaps just confuse the situation if later you _do_ implement remote decoration (using some data structure that is safe in the agent JVM unlike `Run`).

> we also need to combine the condition with `isBuildAgentErrorsEnabled()`

Seems redundant since `of` already runs this check.

> `#of` implementations are executed only on the controller, right?

Right.